bindings/src/main/java/io/pravega/storage/gcp/GCPChunkStorage.java

```diff
@@ -0,0 +1,243 @@
+        }
+    }
+
+    @Override
+    protected boolean checkExists(String chunkName) {
+        Page<Blob> blobs = this.storage.list(this.config.getBucket(), Storage.BlobListOption.prefix(getObjectPath(chunkName)), Storage.BlobListOption.pageSize(1));
+        return blobs.getValues().iterator().hasNext();
+    }
+
+    @Override
+    protected void doDelete(ChunkHandle handle) throws ChunkNotFoundException {
+        boolean deleted = this.storage.delete(this.config.getBucket(), getObjectPath(handle.getChunkName()));
+        if (!deleted) {
+            throw new ChunkNotFoundException(handle.getChunkName(), "doDelete", null);
+        }
+    }
+
+    /**
+     * This method takes exception instance and converts it to a meaningful exception.
+     * @param chunkName - Name of the chunk
+     * @param message - Error message
+     * @param e - Exception
+     * @return Takes the exception and converts it to meaningful exception
+     */
+    private ChunkStorageException convertException(String chunkName, String message, Exception e) {
+        ChunkStorageException retValue = null;
+        if (e instanceof IOException) {
+            if (e.getCause() instanceof RetryHelper.RetryHelperException) {
+                RetryHelper.RetryHelperException retryHelperException = (RetryHelper.RetryHelperException) e.getCause();
+                if (retryHelperException.getCause() instanceof StorageException) {
+                    StorageException storageException = (StorageException) retryHelperException.getCause();
+                    retValue = getChunkNotFoundException(chunkName, message, storageException);
+                }
+            }
+        } else if (e instanceof IllegalArgumentException) {
+            throw (IllegalArgumentException) e;
+        } else {
+            retValue = new ChunkStorageException(chunkName, message, e);
+        }
+
+        return retValue;
+    }
+
+    /**
+     * This method takes StorageException and converts it to ChunkNotFoundException.
+     * @param chunkName - Name of the chunk
+     * @param message - Error message
+     * @param storageException - StorageException
+     * @return Takes the StorageException and returns ChunkNotFoundException
+     */
+    private ChunkNotFoundException getChunkNotFoundException(String chunkName, String message, StorageException storageException) {
+        int code = storageException.getCode();
+        ChunkNotFoundException chunkNotFoundException = null;
+        if (code == FILE_NOT_FOUND) {
+            chunkNotFoundException = new ChunkNotFoundException(chunkName, message, storageException);
+        }
+        return chunkNotFoundException;
+    }
+
+    /**
+     * This method takes object name and append a prefix to it.
+     * @param objectName - The name of the object
+     * @return returns the object name by appending prefix
+     */
+    private String getObjectPath(String objectName) {
+        return config.getPrefix() + objectName;
```

Comment:
Is this method invoked often? Just be aware of the `+` operation with Strings, which makes copies of Strings for concatenating them.